tests/aws/services/s3/test_s3.py

```diff
@@ -964,31 +978,132 @@ def test_create_bucket_via_host_name(self, s3_vhost_client, aws_client, region_n
             s3_vhost_client.delete_bucket(Bucket=bucket_name)
 
     @markers.aws.validated
-    def test_put_and_get_bucket_policy(self, s3_bucket, snapshot, aws_client, allow_bucket_acl):
```

Comment:
nice refactoring of the tests, this makes it easier to validate all the use-cases now 💯